### DISCUSSION TITLE
How to use material time-step adaptivity functionality?

### DISCUSSION THREAD

**Discussion by mojtabaAmiraslanpour** [UvaCsl/HemoCell#67] [General discussion]
Hi,

I mentioned that using material time-step adaptivity (alongside random cell packing), simulation runtime would decrease significantly [1]. I see that in the config file for the **_pipeflow_** case we set "_stepMaterialEvery_" option which would fix the time step for the material model update. How can I enable the adaptivity function so it changes according to the largest appearing force of the material system?

[1] Závodszky, Gábor, et al. "Hemocell: a high-performance microscopic cellular library." Procedia Computer Science 108 (2017): 159-165.

**Comment by gzavo** [comment]
Hi @mojtabaAmiraslanpour,

Sorry, but the related code is not available in the open-source branch at the moment.
The main reason is that the adaptive code caused some stability issues in more complicated geometries, and so it was only used for performance benchmarks. Replicating some of its functionality is not very difficult if you are interested:  it is possible to change the material and particle step frequencies during the simulation. You can come up with any rule on when and how to change them, just be careful the accumulating error is proportional to the step size.
